codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttPubAckMessage.java

```diff
@@ -1,32 +0,0 @@
-/*
- * Copyright 2014 The Netty Project
- *
- * The Netty Project licenses this file to you under the Apache License,
- * version 2.0 (the "License"); you may not use this file except in compliance
- * with the License. You may obtain a copy of the License at:
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations
- * under the License.
- */
-
-package io.netty.handler.codec.mqtt;
-
-/**
- * See <a href="http://public.dhe.ibm.com/software/dw/webservices/ws-mqtt/mqtt-v3r1.html#puback">MQTTV3.1/puback</a>
- */
-public final class MqttPubAckMessage extends MqttMessage {
```

Comment:
We can't do this as it would be API breakage

Reply:
Restored `MqttPubAckMessage` and its encoding/decoding. It makes little sense, however - PUBACK, PUBREC, PUBREL and PUBCOMP can be handled as `MqttMessage` (which is superclass of `MqttPubAckMessage`), just with a custom variable header class. Does it make sense to mark `MqttPubAckMessage` as deprecated?